strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -712,11 +723,63 @@ public void testLastVersionManagement(TestInfo testInfo)
         artifactEntryWithClassifier = artifactEntryService.findOneArtifact(STORAGE0,
                                                                            repositoryId,
                                                                            artifactPathWithClassifier);
-                                                                           
+
         MatcherAssert.assertThat(artifactEntryWithClassifier.getTagSet(), CoreMatchers.notNullValue());
         MatcherAssert.assertThat(artifactEntryWithClassifier.getTagSet().size(), CoreMatchers.equalTo(0));
     }
 
+    @Test
+    public void testChecksumsStorage(TestInfo testInfo)
+            throws Exception
+    {
+        String repositoryId = getRepositoryName("checksums-storage", testInfo);
+
+        MutableRepository repository = mavenRepositoryFactory.createRepository(repositoryId);
+        repository.setLayout(Maven2LayoutProvider.ALIAS);
+
+        createRepositoryWithArtifacts(STORAGE0,
+                                      repository,
+                                      "org.carlspring.strongbox:strongbox-utils",
+                                      "8.7");
+
+        String gavtc = "org.carlspring.strongbox:strongbox-utils:8.4:jar";
+
+        File repositoryBasedir = getRepositoryBasedir(STORAGE0, repositoryId);
+        generateArtifact(repositoryBasedir.getAbsolutePath(), gavtc);
```

Comment:
you should use this generated adtifact below instead of `generateArtifactInputStream`. 
you also have checksum files generated here, so you can read this checksums and use them as `expectedChecksum` below.

Reply:
> you should use this generated adtifact below instead of `generateArtifactInputStream`. 
@sbespalov 
1. The method `mavenArtifactManagementService.store` requires as parameter a variable of type InputStream. If I remove `generateArtifactInputStream`, I will have no parameter for '`store`' method: `generateArtifact` returns a type MavenArtifact and am not aware of how to get from this type to the type InputStream. Or maybe, in this particular case I do not have to invoke `mavenArtifactManagementService.store` at all?
> you also have checksum files generated here
2. How can I access the checksum generated here?

Reply:
Yes, you need an InputStream, and you can get it from file, which the generateArtifact method produce inside. Try to go inside the method and look what happened there.